wikilink/wiki_link.py

```diff
@@ -525,3 +570,26 @@ def _insert_link(session, from_page_id, to_page_id, no_of_separation):
 			"Many rows found in DB to store link from {} to {}\
 			 with number of seperation {}".format(from_page_id, to_page_id,
 		 								          no_of_separation))
+
+class LinkNode(object):
+    def __init__(self, node_id):
+        self.node_id = node_id
+        self.path = [] #parent urls
+
+def get_all_my_parents(node_id, session):
+    parent_list = []
+    for every_parent in session.query(Link).filter(Link.to_page_id==node_id,Link.number_of_separation==1):
+        parent_node = LinkNode(every_parent.from_page_id)
+        parent_list.append(parent_node)
+    return parent_list
+
+def get_parent_id_list(parent_list):
+    return [parent_node.node_id for parent_node in parent_list]
+
+def get_url_path_from_page_ids(session, page_ids_list):
+    path_url = []
+    for page_id in page_ids_list:
+        page_url = session.query(Page).filter(Page.id == page_id).first()

```

Comment:
For DB operation, please consider to add try/except, i.e. in case DB connection failure